java-design-patterns/src/main/java/org/example/Main.java

```diff
@@ -17,7 +18,29 @@ private static void testSingleton(){
 
 //        SingletonCache singletonCache2 = globalCache.clone()
     }
+
+    private static CalculatorCreator getCalculatorCreator(String os){
```

Comment:
problem with string is it is not bounded. user can send any value. what if u use `enum`?